Wrapping/Python/Packaging/setup.py.in

```diff
@@ -1,84 +0,0 @@
-import sys
-
-if "setuptools" in sys.modules.keys():
-    from setuptools import setup, Distribution
-else:
-    from distutils.core import setup
-    from distutils.dist import Distribution
-
-import re
-
-doc_files = @SimpleITK_DOC_FILES_AS_LIST@
-
-def get_pep386version():
```

Comment:
SimpleITK currently has a very robust scrip written in CMake to extract all versioning number based on the git repository, and the recent tags. It automatically determine post release versioning vs development versioning, and experimental versioning. It is used by the core libraries, and all wrapped languages need to use the same versioning system.

Here is the script: https://github.com/SimpleITK/SimpleITK/blob/master/CMake/sitkSourceVersion.cmake A little work may need to be done to enable it to run stand-alone outside of the SimpleITK project and produce output that Python ( or another language) can consume. But consistent versioning and only one method to extract the version for the SimpleITK repo should justify the needed tweak.

What will be needed from scikit build is a way to execute CMake scripts in the same environment that the configuration/build will occur. The idea would be to write a CMake script which produces the version information ( perhaps in stdout, or a generated python script ). Then write a python function which executes that CMake script, and returns the version of SimpleITK. This function will be used to define the SimpleITK version for setup.py.

Do scikit build expose the functionality to execute CMake scripts?